### PR TITLE
{data}[GCC/5.4.0-2.26] HDF5 v1.8.17

### PR DESCRIPTION
**Pull request by vanzod**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by vanzod** [comment]
Test report by @vanzod
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
cermis - Linux debian 8.9, Intel(R) Xeon(R) CPU           E5620  @ 2.40GHz, Python 2.7.9
See https://gist.github.com/390d516dd19d8dc55492877c052ffc8e for a full test report.

**Comment by boegel** [comment]
@vanzod Up until now we've built this with a full toolchain, do you have a particular reason for building it with a compiler-only toolchain?

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.17-GCC-5.4.0-2.26-serial.eb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+version = '1.8.17'
+versionsuffix = '-serial'
+
+homepage = 'https://support.hdfgroup.org/HDF5/'
+description = """HDF5 is a data model, library, and file format for storing and managing data.
+ It supports an unlimited variety of datatypes, and is designed for flexible
+ and efficient I/O and for high volume and complex data."""
+
+toolchain = {'name': 'GCC', 'version': '5.4.0-2.26'}
+toolchainopts = {'optarch': True, 'pic': True}
```
Comment (boegel) [inline easybuild/easyconfigs/h/HDF5/HDF5-1.8.17-GCC-5.4.0-2.26-serial.eb]: `optarch` is enabled by default, you should set `usempi` to `False` for a serial build?

Comment (vanzod) [reply easybuild/easyconfigs/h/HDF5/HDF5-1.8.17-GCC-5.4.0-2.26-serial.eb]: I am building with `GCC` toolchain and `usempi` is defined only for toolchains with `MPI`.
I will go ahead and clean the `optarch`.

**Comment by vanzod** [comment]
@boegel The reason why I want a serial version is because there are a set of software (e.g. grace, ITK) that depend on it but do not need MPI. And since some of those act as dependencies for something else, that would bring even more software to a higher toolchain.

That said, I can live with it at the `foss` toolchain if you prefer not to merge it.

**Comment by boegel** [comment]
@vanzod I'm not opposing the merge, just wondering.

Also, keep in mind that this may cause problems if you want to combine `ITK` with something else that was built with `foss` and also requires `HDF5`. But that might be academic.

File: easybuild/easyconfigs/h/HDF5/HDF5-1.8.17-GCC-5.4.0-2.26-serial.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/h/HDF5/HDF5-1.8.17-GCC-5.4.0-2.26-serial.eb]: Good point, sorry about missing that.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (1 easyconfigs in this PR)
node2532.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/8cb6e5a04a70fb8eace7fe1f195d8d02 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2072.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/a3750db38bc2cd8ca5d801ec1cb687e6 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @vanzod!

**Pull request merged**